pkg/inventory/inventory-client.go

```diff
@@ -33,7 +34,7 @@ type InventoryClient interface {
 	Merge(inv InventoryInfo, objs object.ObjMetadataSet, dryRun common.DryRunStrategy) (object.ObjMetadataSet, error)
 	// Replace replaces the set of objects stored in the inventory
 	// object with the passed set of objects, or an error if one occurs.
-	Replace(inv InventoryInfo, objs object.ObjMetadataSet, dryRun common.DryRunStrategy) error
+	Replace(inv InventoryInfo, objs object.ObjMetadataSet, dryRun common.DryRunStrategy, applyStatus map[object.ObjMetadata]status.ApplyReconcileStatus) error
```

Comment:
Is the set of keys in the `applyStatus` map always the same as the set of `objs` here?

Reply:
I think it should be the same as the `objs` set, but I haven't verified it.